nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/MemberAddressPicker.java

```diff
@@ -97,52 +109,78 @@ public InetSocketAddress getBindAddress() {
-                publicAddress = new InetSocketAddress(address[0], Integer.parseInt(address[1]));
-            } else {
-                publicAddress = new InetSocketAddress(address[0], Integer.parseInt(config.getStartPort()));
-            }
-        }
-        
+
+    private InetSocketAddress initPublicAddress(InetSocketAddress bindAddress) {
         logger.fine("Finding an appropriate address for Hazelcast to use");
+        InetSocketAddress address;
         int port = 0;
         if (env.isDas() && !env.isMicro()) {
             port = new Integer(config.getDasPort());
-            if (config.getDASPublicAddress() != null && !config.getDASPublicAddress().isEmpty()) {
-                publicAddress = new InetSocketAddress(config.getDASPublicAddress(), port);
+            // try setting public address from global Payara config
+            address = initAddress(config.getDASPublicAddress(), port);
+        } else {
+            // try setting public address from configuration, Payara node config
+            address = initAddress(localConfig.getPublicAddress(), Integer.parseInt(config.getStartPort()));
+        }
+        return ensureAddress(address, bindAddress, chosenAddress, port);
+    }
+
+    static InetSocketAddress initAddress(String address, int port) {
+        if (address != null && !address.isEmpty()) {
+            String addressParts[] = address.split(":");
+            if (addressParts.length > 1) {
+                return new InetSocketAddress(addressParts[0], Integer.parseInt(addressParts[1]));
+            } else {
+                return new InetSocketAddress(addressParts[0], port);
             }
-            
-            if (config.getDASBindAddress() != null && !config.getDASBindAddress().isEmpty()) {
-                bindAddress = new InetSocketAddress(config.getDASBindAddress(), port);
-                logger.log(Level.FINE, "Bind address is specified in the configuration so we will use that {0}", bindAddress);
-                return;
+        } else {
+            return null;
+        }
+    }
+
+    private static InetSocketAddress ensureAddress(InetSocketAddress targetAddress, InetSocketAddress sourceAddress,
+            LazyHolder<InetAddress> backupAddress, int port) {
+        if (targetAddress == null) {
+            if (sourceAddress != null && !sourceAddress.getAddress().isAnyLocalAddress()) {
+                targetAddress = sourceAddress;
+            } else if (backupAddress.get() != null) {
+                targetAddress = new InetSocketAddress(backupAddress.get(), port);
+            } else {
+                targetAddress = tryLocalHostOrLoopback(port);
             }
         }
-       
-        
+        return targetAddress;
+    }
+
+    private static InetSocketAddress tryLocalHostOrLoopback(int port) {
+        try {
+            // ok do the easy thing
+            logger.log(Level.FINE, "Could not find an appropriate address by searching falling back to local host");
```

Comment:
Slight wording change - the "by searching" bit is unnecessary and also irks me because you never say what or where you were searching 😜
```suggestion
            logger.log(Level.FINE, "Could not find an appropriate address, falling back to local host");
```